rstream/client.py

```diff
@@ -201,22 +213,30 @@ async def _listener(self) -> None:
         assert self._conn
         while True:
             try:
-                frame = await self._conn.read_frame()
+                if self._conn is not None:
+                    frame = await self._conn.read_frame()
```

Comment:
Same as the first comment _is_not_closed was just a special flag to notify _run_delivery_handlers that there wa s a clean closing. (maybe a bit confusing and we may thing to rename it). We could introduce a new flag for when connections are forcely closed or just use _conn and set it to None in case of an unexpected connection close.

Reply:
Implemented as the first suggestion  _is_not_closed is now a @property and can be used also for this situation